armory/environment.py

```diff
@@ -2,4 +2,243 @@
 Environment parameter names
 """
 
+# TODO This is a relic...remove
 ARMORY_VERSION = "ARMORY_VERSION"
+
+from pydantic import BaseModel, validator
+import os
+from armory.logs import log
+
+
+class EnvironmentConfiguration(BaseModel):
+    """Armory Environment Configuration Context
+
+    This Dataclass contains the environmental references
+    necessary for armory execution.
+    """
+
+    ARMORY_CONFIGURATION_DIRECTORY: str = os.path.expanduser("~")
```

Comment:
because these objects only exist in an EnvironmentConfiguration, the `ARMORY_` part is redundant and adds verbosity without adding information

Reply:
And ALL_CAPS is tolerable in global (module-scope) variables, but wrong as a member name.

It also hurts my eyes and brain.